.github/workflows/selfhosted.yml

```
@@ -54,9 +54,30 @@ jobs:
         conda create --prefix=testenv python=${{ matrix.python-version }}
         conda activate ./testenv
         python -m pip install --upgrade pip
-        pip install pytest
+        pip install pytest-cov
         pip install git+https://github.com/qiboteam/qibo
         pip install .[tiiq]
     - name: Test ${{ matrix.platform }}
       run: |
-        srun -p ${{ matrix.platform }} pytest src/qibolab --pyargs qibolab -m qpu --platform ${{ matrix.platform }} -rx
+        # create script for slurm
+        platform=${{ matrix.platform }}
+        echo \#\!/bin/bash >> ci_tests
+        echo >> ci_tests
+        echo "mkdir /tmp/qibolab" >> ci_tests
+        echo "ln -sr src/qibolab/tests /tmp/qibolab" >> ci_tests
+        echo "cd /tmp/qibolab" >> ci_tests
+        echo "pytest --cov=qibolab --cov-report=xml --platform $platform -rx" >> ci_tests
+        echo "cd -" >> ci_tests
+        echo "mv /tmp/qibolab/coverage.xml ." >> ci_tests
+        echo "rm -r /tmp/qibolab/" >> ci_tests
+        # execute script in the queue
+        chmod +x ci_tests
+        srun -p ${{ matrix.platform }} ci_tests
```

```suggestion
        sed -i 's/$PLATFORM/'${{ matrix.platform }}'/' my_slurm.bash
        srun -p ${{ matrix.platform }} my_slurm.bash
```

(the `sed` command has to be tested and possibly debugged)

---

Many thanks for this suggestion, it works and is much much cleaner than what I was doing. The only thing is that I placed the script in the rootdir so that I do not need to use longer dirs in the workflow, but if you think it is better somewhere else (eg. within the .github folder, we can move it).